### PR TITLE
Java: Adding command `LCS` (No IDX option)

### PR DESCRIPTION
**Pull request by tjzhang-BQ**
This only covers the signature where LCS returns the LCS as a `String` and LCS length as a `Long`. `Map` return type with the IDX option is coming as a part II.

Edit:
`MINMATCHLEN` does not seem to work:
```
127.0.0.1:6379> set a1 abc
OK
127.0.0.1:6379> set a2 bcd
OK
127.0.0.1:6379> lcs a1 a2
"bc"
127.0.0.1:6379> lcs a1 a2 len
(integer) 2
127.0.0.1:6379> lcs a1 a2 MINMATCHLEN 3
"bc"
127.0.0.1:6379> lcs a1 a2 len MINMATCHLEN 3
(integer) 2
127.0.0.1:6379> lcs a1 a2 idx MINMATCHLEN 3
1) "matches"
2) (empty array)
3) "len"
4) (integer) 2
```

And as a reference https://github.com/valkey-io/valkey/pull/420 discusses how `WITHMATCHLEN` and `MINMATCHLEN` only works/makes sense with the `IDX` option but is still allowed to be added when `IDX` isn't present. 

For now, this PR will be excluding the `MINMATCHLEN` option since it only includes the default behavior without optional keywords and the `LEN` option behavior, which is mutually exclusive with `IDX`.

### CODE REVIEW

**Review by yipin-chen — APPROVED** [review]

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 1]
```diff
@@ -316,4 +317,47 @@ public interface StringBaseCommands {
...
+     * @param options optional input that indicates the length of the lcs should be returned.
+     * @return The length of the longest common subsequence between the 2 strings.
+     * @example
+     *     <pre>{@code
+     * client.set("testKey1", "abcd");
+     * client.set("testKey2", "axcd");
+     * assertEquals(3L, client.lcs("testKey1", "testKey2", new LCSOptions(true));
+     * }</pre>
+     */
+    CompletableFuture<Long> lcs(String key1, String key2, LCSOptions options);
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:362]: What will be the signature of `lcs` with `IDX` option?
```
CompletableFuture<Map<?, ?>> lcs(String key1, String key2, LCSOptions options);
```
is causes a conflict. Please think about it now and post it to PR description.
It is also OK to add it scope of this PR with dummy implementation which just throws NotImplementedException.

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 2]
```diff
@@ -316,4 +317,47 @@ public interface StringBaseCommands {
...
+     * Returns the length of the longest common subsequence between strings stored at <code>key1
+     * </code> and <code>key2</code>.
+     *
+     * @since Redis 7.0 and above.
+     * @apiNote When in cluster mode, <code>key1</code> and <code>key2</code> must map to the same
+     *     hash slot.
+     * @see <a href="https://valkey.io/commands/lcs/">valkey.io</a> for details.
+     * @param key1 The key that stores the first string.
+     * @param key2 The key that stores the second string.
+     * @param options optional input that indicates the length of the lcs should be returned.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:353]: Not optional for this function
```suggestion
     * @param options Additional parameters ...
```

File: java/client/src/main/java/glide/api/models/commands/LCSOptions.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    private final boolean isLEN;
+
+    /** The minimal match length value, <code>-1</code> if the option is not included */
+    private long minMatchLen = -1;
+
+    /** Indicates if the WITHMATCHLEN option should be included */
+    private final boolean withMatchLen;
+
+    /** Optional arguments constructor with the Length/IDX option. */
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/LCSOptions.java:42]:
```suggestion
    /** Optional arguments constructor with the Length option. */
```

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 4]
```diff
@@ -316,4 +317,47 @@ public interface StringBaseCommands {
...
+     * @return The length of the longest common subsequence between the 2 strings.
+     * @example
+     *     <pre>{@code
+     * client.set("testKey1", "abcd");
+     * client.set("testKey2", "axcd");
+     * assertEquals(3L, client.lcs("testKey1", "testKey2", new LCSOptions(true));
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:359]:
```suggestion
     * assertEquals(3L, client.lcs("testKey1", "testKey2", new LCSOptions(true).get());
```

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 5]
```diff
@@ -316,4 +317,47 @@ public interface StringBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> append(String key, String value);
+
+    /**
+     * Returns the longest common subsequence between strings stored at <code>key1</code> and <code>
+     * key2</code>.
+     *
+     * @since Redis 7.0 and above.
+     * @apiNote When in cluster mode, <code>key1</code> and <code>key2</code> must map to the same
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:326]: <3

**Review by Yury-Fridlyand — COMMENTED** [review]
Redis docs are incorrect, becase `MINMATCHLEN` does not work without `IDX`. Is there is a bug in redis reported on their repo?

```
127.0.0.1:6379> set a1 abc
OK
127.0.0.1:6379> set a2 bcd
OK
127.0.0.1:6379> lcs a1 a2
"bc"
127.0.0.1:6379> lcs a1 a2 len
(integer) 2
127.0.0.1:6379> lcs a1 a2 MINMATCHLEN 3
"bc"
127.0.0.1:6379> lcs a1 a2 len MINMATCHLEN 3
(integer) 2
127.0.0.1:6379> lcs a1 a2 idx MINMATCHLEN 3
1) "matches"
2) (empty array)
3) "len"
4) (integer) 2
```

Please post the valid command signature to the PR, so we can't understand what is going on.
You need also update your options class to reflect that.

File: java/client/src/main/java/glide/api/models/commands/LCSOptions.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    public LCSOptions(boolean isLEN) {
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/LCSOptions.java:43]: Maybe we need to use a factory methods or a builder. I can't analyze what is better now, becase I don't know how command works (redis docs are misleading)

File: java/client/src/main/java/glide/api/models/commands/LCSOptions.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    /**
+     * Converts LCSOptions into a String[].
+     *
+     * @return String[]
+     */
+    public String[] toArgs() {
+        if (isLEN) {
+            return new String[] {LENGTH_REDIS_API};
+        } else {
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/LCSOptions.java:63]: Probably
```
else if (isIDX)
```

File: java/client/src/main/java/glide/api/models/commands/LCSOptions.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        this.isLEN = false;
+        this.minMatchLen = minMatchLen;
+        this.withMatchLen = withMatchLen;
+    }
+
+    /**
+     * Converts LCSOptions into a String[].
+     *
+     * @return String[]
+     */
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/LCSOptions.java:59]: COnsider adding UT for this class

Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/LCSOptions.java:59]: COnsider adding UT for this class

File: java/client/src/main/java/glide/api/BaseClient.java [hunk 9]
```diff
@@ -1786,4 +1787,16 @@ public CompletableFuture<Boolean> copy(@NonNull String source, @NonNull String d
...
     }
+
+    @Override
+    public CompletableFuture<String> lcs(@NonNull String key1, @NonNull String key2) {
+        String[] arguments = new String[] {key1, key2};
+        return commandManager.submitNewCommand(LCS, arguments, this::handleStringResponse);
+    }
+
+    @Override
+    public CompletableFuture<Long> lcsLEN(@NonNull String key1, @NonNull String key2) {
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/BaseClient.java:1798]: Rename to `lcsLen`

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 10]
```diff
@@ -316,4 +319,47 @@ public interface StringBaseCommands {
...
+     * @see <a href="https://valkey.io/commands/lcs/">valkey.io</a> for details.
+     * @param key1 The key that stores the first string.
+     * @param key2 The key that stores the second string.
+     * @return A <code>String</code> containing the longest common subsequence between the 2 strings.
+     *     An empty <code>String</code> is returned if the keys do not exist or have no common
+     *     subsequences.
+     * @example
+     *     <pre>{@code
+     * client.set("testKey1", "abcd");
+     * client.set("testKey2", "axcd");
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:339]: You can omit those. Or if you want to keep - fix.

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 11]
```diff
@@ -316,4 +319,47 @@ public interface StringBaseCommands {
...
+     * assertEquals("acd", client.lcs("testKey1", "testKey2").get());
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:340]: `assertEquals` and other assert functions are designed for tests.
```suggestion
     * String result = client.lcs("testKey1", "testKey2").get();
     * assert result.equals("acd");
```

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 12]
```diff
@@ -316,4 +319,47 @@ public interface StringBaseCommands {
...
+     *     hash slot.
+     * @see <a href="https://valkey.io/commands/lcs/">valkey.io</a> for details.
+     * @param key1 The key that stores the first string.
+     * @param key2 The key that stores the second string.
+     * @param len Indicates if the
+     * @return The length of the longest common subsequence between the 2 strings.
+     * @example
+     *     <pre>{@code
+     * client.set("testKey1", "abcd");
+     * client.set("testKey2", "axcd");
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:360]: same

File: java/client/src/main/java/glide/api/models/commands/LcsIdxOptions.java [hunk 13]
```diff
@@ -0,0 +1,82 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import lombok.Getter;
+import org.apache.commons.lang3.ArrayUtils;
+
+/**
+ * Optional arguments for TOBEIMPLEMENTED command.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/LcsIdxOptions.java:10]: You can delete this file now, and restore later, when you'll work on TOBEIMPLEMENTED command

**Review by acarbonetto — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]
I unresolved two comments from previous review

**Pull request merged**